.github/pull_request_template.md

```diff
@@ -0,0 +1,41 @@
+## Describe your changes
+< Summary of the changes.>
+
+< Please also include relevant motivation and context. >
+
+< List any dependencies that are required for this change. >
+
+## Type of change
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+
+## Checklist before requesting a review
+
+- [ ] My branch is up-to-date with the target branch - if not update your fork with the changes from the target branch (use `pull` with `--rebase` option if possible).
+- [ ] I have performed a self-review of my code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] I have requested a reviewer and an assignee (assignee is responsible for merging)
+
+## Checklist for reviewers
+
+Each PR comes with its own improvements and flaws. The reviewer should check the following:
+- [ ] the code readable
+- [ ] the code well tested
+- [ ] the code documented
+- [ ] the code easy to maintain
+
+## Author checklist after completed review
+
+- [ ] I have added a line to the CHANGELOG describing this change (in section
+  reflecting type of change, for example "bug fixes", add section where
+  missing)
```

Comment:
> I guess a drawback with just editing the changelog immediately is that what should be written there could change after discussions in the PR, but I think that is quite rare, and then you just have to remember to update the changelog edit as well.

Yes, to me this is the key drawback. Sometimes people will want to fix/change/add something and they way they have achieved that might change as the review process progresses. In my experience this is often the case. Also, people often forget to update the changelog so this is a a little last-minute reminder to do that before merging. And finally, a given PR doesn't have a URL until it has been created, and so updating the changelog with a separate commit after the PR as been created is typically necessary.

Reply:
> A question about the changelog sections: This here seems to apply that one should add arbitrary new section if there is not a new one? But below the changelog sections are described as fixed (_added_, _changed_, _fixed_). What is the general idea for this? If we are sticking to explicitly the sections listed at https://keepachangelog.com/en/1.1.0/, then I think we should make that clear also here. One way to do that is to always have those sections under _unreleased_, and then there is never a need to add sections with a PR. Empty sections at the time of a release can just be removed.

Yes, I think having fixed sections sounds like a good idea. I was going to suggest we discuss what these should be so that we are on the same page about it. One section that might be good to include (in addition to *added, changed, fixed*) is *maintenance* (these are things that don't change fhe functionality of the code at all, but are say necessary package updates because something breaks, fixing the ci/cd linting, testing etc update). We can also put those in the other sections since they will often *fix* something, but sometimes this info isn't relevant to you if you just want to decide whether to update or not.

Reply:
I'm ok with keeping it like this then. Just make the changelog categories explicit and consistent in the template, _added, changed, fixed, maintenance_ sound good to me.